src/main/java/Airport.java

```diff
@@ -0,0 +1,137 @@
+        for (Plane plane : planesList) {
+            if (plane instanceof PassengerPlane) {
+                passengersPlaneList.add((PassengerPlane) plane);
+            }
+        }
+        return passengersPlaneList;
+    }
+
+    public List<MilitaryPlane> getMilitaryPlanes() throws NonMilitaryPlainException {
+        List<MilitaryPlane> militaryPlanes = new ArrayList<MilitaryPlane>();
+        for (Plane plane : planes) {
+            if (plane instanceof MilitaryPlane) {
+                militaryPlanes.add((MilitaryPlane) plane);
+            } else {
+                throw new NonMilitaryPlainException("Plane is not military");
+            }
+        }
+        return militaryPlanes;
+    }
+
+    public PassengerPlane getPassengerPlaneWithMaxPassengersCapacity() {
+        List<PassengerPlane> passengerPlanes = getPassengersPlane();
+        PassengerPlane planeWithMaxCapacity = passengerPlanes.get(0);
+        for (int i = 0; i < passengerPlanes.size(); i++) {
+            if (passengerPlanes.get(i).getPassengersCapacity() > planeWithMaxCapacity.getPassengersCapacity()) {
+                planeWithMaxCapacity = passengerPlanes.get(i);
+            }
+        }
+        return planeWithMaxCapacity;
+    }
+
+    public List<MilitaryPlane> getTransportMilitaryPlanes() throws NonMilitaryPlainException {
+        List<MilitaryPlane> militaryPlanes = getMilitaryPlanes();
+        List<MilitaryPlane> transportMilitaryPlanes = new ArrayList<>();
+        for (int i = 0; i < militaryPlanes.size(); i++) {
+            MilitaryPlane plane = militaryPlanes.get(i);
+            if (plane.getType() == MilitaryType.TRANSPORT) {
+                transportMilitaryPlanes.add(plane);
+            }
+        }
+        return transportMilitaryPlanes;
+    }
+
+    public List<MilitaryPlane> getBomberMilitaryPlanes() throws NonMilitaryPlainException {
+        List<MilitaryPlane> militaryPlanes = getMilitaryPlanes();
+        List<MilitaryPlane> bomberMilitaryPlanes = new ArrayList<>();
+        for (int i = 0; i < militaryPlanes.size(); i++) {
+            MilitaryPlane plane = militaryPlanes.get(i);
+            if (plane.getType() == MilitaryType.BOMBER) {
+                bomberMilitaryPlanes.add(plane);
+            }
+        }
+        return bomberMilitaryPlanes;
+
+    }
+
+    public List<ExperimentalPlane> getExperimentalPlanes() {
+        List<ExperimentalPlane> experimentalPlanes = new ArrayList<>();
+        for (Plane plane : planes) {
+            if (plane instanceof ExperimentalPlane) {
+                experimentalPlanes.add((ExperimentalPlane) plane);
+            }
+        }
+        return experimentalPlanes;
+    }
+
+    public Airport sortByMaxDistance() {
+        Collections.sort(planes, new Comparator<Plane>() {
+            public int compare(Plane o1, Plane o2) {
+                return o1.GetMaxFlightDistance() - o2.GetMaxFlightDistance();
+            }
+        });
+        return this;
```

Comment:
It better to copy original collection, sort in and return. same for all sort methods in the class